composer.json

```diff
@@ -9,11 +9,11 @@
 		"clue/block-react": "^1.4",
 		"clue/ndjson-react": "^1.0",
 		"composer/ca-bundle": "^1.2",
+		"composer/package-versions-deprecated": "^1.11.0",
```

Comment:
Why would we need this? We can rely on the PHAR being generated with Composer 2.

Reply:
sorry I have no clue how the phar creation process exactly works and misunderstood your last comment obviously. this means changes are needed in the scoper config, right? how does it work, would the needed classes then be copied in to the phar? I don't quite understand the details, would be greate if you can maybe just point me to some examples or so if you know more already

Reply:
Call `InstalledVersions` class that's part of `vendor/composer`?

Reply:
this class is only available in composer 2.x

the dependency added here brings the same functionality für composer 1.x.

the composer 1.x stuff might not be necessary because of the phpstan build process though

Reply:
PHPStan's PHAR is always built with Composer 2.

Reply:
We can also test that PHPStan works when installed with Composer 1 in https://github.com/phpstan/phpstan/blob/1.8.x/.github/workflows/other-tests.yml.

Reply:
brought back the composer 2 changes from yesterday. the error on phar creation is
> PHP Fatal error:  Uncaught Error: Class "_PHPStan_2a31e4f7c\Composer\InstalledVersions" not found in phar:///home/runner/work/phpstan-src/phpstan-src/tmp/phpstan.phar/bin/phpstan:117

this is why I'm constantly talking about it and some scoper config I don't know anything about :) not sure how that class (and its deps?) end up in the phar so that everyone can use it

Reply:
Yeah, because PHP-Scoper isn't perfect, you need to write a similar patch callback as it's here https://github.com/phpstan/phpstan-src/blob/cc7bbc3c7c7f4a259273be975ead34686ee1f215/compiler/build/scoper.inc.php#L172-L184.